statemachine_doc.go

```diff
@@ -100,6 +105,43 @@ func (sm *stateMachine) DescribeTransitionType(transitionType TransitionType, tr
 	sm.transitionTypeDocs[transitionType] = transitionTypeDocumentation
 }
 
+// AsDotGraph identifies the nodes and edges in the transitionrules and
+// returns a *dot.Graph type from the package https://github.com/emicklei/dot.
+//
+// The returned graph can then be rendered into the Graphviz dot form
+// by invoking .String() on the returned *dot.Graph value.
+func (sm *stateMachine) AsDotGraph() *dot.Graph {
+	g := dot.NewGraph(dot.Directed)
+
+	// unique node map
```

Comment:
Suggesting to add comments only in case the code is not clear, in this case no need for the comment